pkg/cmd/cli/cmd/expose.go

```diff
@@ -132,7 +132,9 @@ func validate(cmd *cobra.Command, f *clientcmd.Factory, args []string) error {
 			cmd.Flags().Set("generator", generator)
 			fallthrough
 		case "route/v1":
-			route, err := cmdutil.UnsecuredRoute(kc, namespace, info.Name, info.Name, kcmdutil.GetFlagString(cmd, "port"))
+			// The upstream generator will chose service.Port instead of service.TargetPort
```

Comment:
wil choose ... for the route TargetPort when there is no port set.  Passing true forces a port always to be populated so the upstream default is not used.